src/blocks/DetailsBlock/DetailsBlock.astro

```diff
@@ -0,0 +1,45 @@
+---
+import type { DetailsBlockFragment } from '@lib/types/datocms';
+
+interface Props {
+  block: DetailsBlockFragment
+}
+
+const { block } = Astro.props;
+const splitChar = '|';
+const hasMultipleValues = (value: string) => value.includes(splitChar);
+const splitValues = (value: string) => value.split(splitChar);
+---
+
+<dl class:list={['details-block', `details-block--${block.display}`]}>
```

Comment:
Please use `block.layout`. See other components and the docs about naming: https://github.com/voorhoede/head-start/blob/main/docs/cms-content-modelling.md